addons/DistancePlus/DistancePlus.lua

```diff
@@ -0,0 +1,392 @@
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of DistancePlus nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Sammeh BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'DistancePlus'
+_addon.author = 'Sammeh'
+_addon.version = '1.3.0.1'
+_addon.command = 'dp'
+
+
+require('tables')
+
+res = require 'resources'
+jadistance = require('ja_distance')
+config = require('config')
+texts = require('texts')
+
+defaults = {}
+defaults.pos = {}
+defaults.pos.x = -178
+defaults.pos.y = 21
+defaults.text = {}
+defaults.text.font = 'Arial'
+defaults.text.size = 14
+defaults.flags = {}
+defaults.flags.right = true
+
+pettxt = {}
+pettxt.pos = {}
+pettxt.pos.x = -178
+pettxt.pos.y = 45
+pettxt.text = {}
+pettxt.text.font = 'Arial'
+pettxt.text.size = 14
+pettxt.flags = {}
+pettxt.flags.right = true
+
+abilitytxt = {}
+abilitytxt.pos = {}
+abilitytxt.pos.x = -80
+abilitytxt.pos.y = 45
+abilitytxt.text = {}
+abilitytxt.text.font = 'Arial'
+abilitytxt.text.size = 10
+abilitytxt.flags = {}
+abilitytxt.flags.right = true
+
+heighttxt = {}
+heighttxt.pos = {}
+heighttxt.pos.x = -238
+heighttxt.pos.y = 21
+heighttxt.text = {}
+heighttxt.text.font = 'Arial'
+heighttxt.text.size = 14
+heighttxt.flags = {}
+heighttxt.flags.right = true
+height_upper_threshold = 8.5
+height_lower_threshold = -7.5
+
+distance = texts.new('${value||%.2f}', defaults)
+petdistance = texts.new('${value||%.2f}', pettxt)
+abilities = texts.new('${value}', abilitytxt)
+height = texts.new('${value||%.2f}', heighttxt)
+
+debug.setmetatable(nil, {__index = {}, __call = functions.empty})
+
+option = "Default"
+showabilities = false
+showheight = false
+
+self = windower.ffxi.get_player()
+
+
+function displayabilities(distance,master_pet_distance)
+  abilitylistrecast = windower.ffxi.get_ability_recasts() 
```

Comment:
Cleaning up thanks. Actually wasn't using the ability recast at all so that pointed it out. That was an idea in motion that never took flight until really another addon.